lib/linux_admin/scap.rb

```diff
@@ -44,28 +54,29 @@ def lockdown_profile(xccdf_file_path, profile_id)
 
     private
 
-    def self.xccdf_file
-      local_ssg_file("xccdf")
+    def self.xccdf_file(platform)
+      local_ssg_file("xccdf", platform)
     end
 
-    def self.oval_file
-      local_ssg_file("oval")
+    def self.oval_file(platform)
+      local_ssg_file("oval", platform)
     end
 
-    def self.local_ssg_file(type)
-      Dir.glob(SSG_XML_PATH.join("ssg-*-#{type}.xml")).detect { |f| f =~ /ssg-\w+-#{type}.xml/ }
+    def self.local_ssg_file(type, platform)
+      platform ||= "*"
+      Dir.glob(SSG_XML_PATH.join("ssg-#{platform}-#{type}.xml")).detect { |f| f =~ /ssg-\w+-#{type}.xml/ }
     end
 
     def tempdir
       @tempdir ||= Pathname.new(Dir.tmpdir)
     end
 
     def xccdf_file
-      @xccdf_file ||= self.class.xccdf_file
+      @xccdf_file ||= self.class.xccdf_file(platform)
```

Comment:
If the platform changes via the accessor, then this cache should be cleared, right?  (same with oval_file)

Reply:
Alternately, use attr_reader instead, then enforce that callers must set the platform on initialize.

Reply:
I'm for the `attr_reader` approach, I don't see someone reusing this object for different platforms so I think it makes sense to set it on initialization.